### ISSUE TITLE
D400 with open3d

### ISSUE THREAD

**Issue by nawara72** [IntelVCL/Open3D#670]
Hi
Does open3d 0.4 work with the D400 series (ie realsenselib2) or only the old realsenselib?

 I tried open3d examples with the D435, but couldn't get it to work?
Thanks
Nawar

**Comment by yxlao** [comment]
Yes, `realsense2` is supported. 

In theory, Open3D is able to support any 3D sensors. All we need to do is to get the data from the sensor and feed to Open3D with the correct format.  So it's more like "example usage" rather than "build-in support". 

We have a D435 demo at https://www.youtube.com/watch?v=6FkbxLLZyFQ. Also check https://github.com/IntelVCL/Open3D/tree/master/examples/Python/ReconstructionSystem/sensors for the code.

**Comment by nawara72** [comment]
Hi yxlao, thanks so much for your reply. I tried to run it and thats what I got
 python.exe .\realsense_pcd_visualizer.py
Traceback (most recent call last):
  File ".\realsense_pcd_visualizer.py", line 29, in <module>
    pipeline = rs.pipeline()
RuntimeError: API version mismatch: librealsense.so was compiled with API version 2.15.0 but the application was compile
d with 2.16.1! Make sure correct version of the library is installed (make install)
PS C:\Program Files\Open3D-0.4.0\examples\Python\ReconstructionSystem\sensors>

Any idea what i am doing wrong?

Thanks

**Comment by yxlao** [comment]
Line 29 is 
```python
pipeline = rs.pipeline()
```
which is pure RealSense code, which is not related to Open3D. I would suggest following the RealSense documentation to resolve this issue.

**Comment by syncle** [comment]
It looks like the version issue. Can you upgrade your realsense library and try again? You can also upgrade the firmware for the camera.

**Comment by nawara72** [comment]
I have updated the firmware. I think the problem is that open3d/3rparty/ librealsense has the old librealsense. How can I compile open3d using the librealsense2, which i have already compiled and installed.?

**Comment by syncle** [comment]
librealsense in our 3rdparty is just provided for reference, it is not the newest one. Please uninstall it and follow http://open3d.org/docs/tutorial/ReconstructionSystem/capture_your_own_dataset.html 
@yxlao we will need to remove librealsense in our repo (or we need to move it to another repo for 3rdparty as we have discussed)

**Comment by syncle** [comment]
Umm I don't know what librealsense2 you are referring (are you using ROS package?). I mentioned that you will need to follow https://github.com/IntelRealSense/librealsense/tree/master/wrappers/python as shown in the [documentation](http://open3d.org/docs/tutorial/ReconstructionSystem/capture_your_own_dataset.html)

**Comment by nawara72** [comment]
I have downloaded librealsense from https://github.com/IntelRealSense/librealsense, compiled, installed and ran the examples with D435 and all works fine. I compiled and installed open3d,
I have copied open3d.cp37-win_amd64.pyd and pyrealsense2.cp37-win_amd64.pyd to same folder of realsense_pcd_visualizer.py
 however can't seem to run the  realsense_pcd_visualizer  with D435, i keep getting the 
pipeline = rs.pipeline()
RuntimeError: API version mismatch: librealsense.so was compiled with API version 2.15.0 but the application was compiled with 2.16.1! Make sure correct version of the library is installed (make install)
Would appreciate any ideas on how to get it to work. Not sure also why it metions librealsense.so and I am running on a windows platform
